pkg/scheduler/actions/allocate/allocate.go

```diff
@@ -62,12 +69,37 @@ func (alloc *allocateAction) Execute(ssn *framework.Session) {
 			continue
 		}
 
+		// ignore namespace order enabled or not, just add key in jobsMap
 		if _, found := jobsMap[job.Queue]; !found {
 			jobsMap[job.Queue] = util.NewPriorityQueue(ssn.JobOrderFn)
 		}
 
-		glog.V(4).Infof("Added Job <%s/%s> into Queue <%s>", job.Namespace, job.Name, job.Queue)
-		jobsMap[job.Queue].Push(job)
+		if !namespaceOrderEnabled {
+			glog.V(4).Infof("Added Job <%s/%s> into Queue <%s>", job.Namespace, job.Name, job.Queue)
+			jobsMap[job.Queue].Push(job)
+			continue
+		}
+
+		if _, found := namespaceMap[job.Queue]; !found {
+			namespaceMap[job.Queue] = util.NewPriorityQueue(ssn.NamespaceOrderFn)
+		}
+
+		namespaceJob, found := jobInNamespaceMap[job.Queue]
+		if !found {
+			namespaceJob = make(map[string]*util.PriorityQueue)
+			jobInNamespaceMap[job.Queue] = namespaceJob
+		}
+
+		jobs, found := namespaceJob[job.Namespace]
+		if !found {
+			jobs = util.NewPriorityQueue(ssn.JobOrderFn)
+			namespaceJob[job.Namespace] = jobs
+
+			namespaceMap[job.Queue].Push(job.Namespace)
+		}
+
+		glog.V(4).Infof("Added Job <%s/%s> into Queue <%s> Namespace <%s>", job.Namespace, job.Name, job.Queue, job.Namespace)
```

Comment:
nit: duplicate namespace

Reply:
change duplicate namespace log to prefix `NamespaceEnabled: ` to indicate this add action is different to adding job directly into map belonging to queue.

Reply:
after code refactoring, this log is deleted.